### DISCUSSION TITLE
Best way to use signOut() but also clear a cookie first?

### DISCUSSION THREAD

**Discussion by transformmindgarden** [nextauthjs/next-auth#5577] [Help]
What I'd like to do is, when the user clicks the sign out button, delete a cookie that's not a next auth cookie, then trigger the built in signOut() call.  

Is there a better way than just putting the delete in the header.tsx before the signOut() call? could I wrap the signOut call so when its called after being imported it first deletes the cookie then continues on?  When I tried to wrap the signOut function I got  ReferenceError: window is not defined suggesting I was calling signOut server side? Thanks for any advice you can provide!

**Comment by aaliyan1230** [comment]
@transformmindgarden i have a similar case too, did you happen to figure out a solution for this?
